### PR TITLE
[JENKINS-34474] New parent pom (2.7)

### PR DESCRIPTION
**Pull request by armfergom**
Superseeds (and includes) #97 

Work included:

- [x] Upgrade to parent pom 2.7
- [x] Fix test errors
- [x] Fix javadoc errors
- [x] Update tests to use `JenkinsRule` instead of `HudsonTestCase`
- [x] Bump baseline to 1.596.1. Reason [here](https://cloudbees.zendesk.com/hc/en-us/articles/204170674-After-updating-Matrix-Project-Plugin-Jenkins-fails-to-restart-)

@reviewbybees @kwhetstone

### CODE REVIEW

**Comment by kwhetstone** [comment]
:bee: Great changes!  This was a really difficult upgrade.  I'm for closing #97 since it was supposed to be a temporary change for a quick turnaround to get to the parent pom & not change too much in the plugin itself.

File: pom.xml [hunk 1]
```diff
@@ -83,86 +92,31 @@
...
-            </execution>
-          </executions>
-        </plugin>
-      </plugins>
-        <pluginManagement>
-            <plugins>
-                <!-- Ignore animal-sniffer plugin when run with m2e on Eclipse. -->
-                <plugin>
-                    <groupId>org.eclipse.m2e</groupId>
-                    <artifactId>lifecycle-mapping</artifactId>
```
Comment (recena) [inline pom.xml]: Thanks for remove this.

File: src/main/java/hudson/plugins/parameterizedtrigger/BuildTriggerConfig.java [hunk 2]
```diff
@@ -154,12 +154,12 @@ public boolean getTriggerWithNoParameters() {
...
-     * The second set contains dynamically configured project, resolved by back tracking builds environment variables.
-     * The third set contains other recently triggered project found during back tracking builds
-     * The fourth set contains dynamically configured project that couldn't be resolved or project that doesn't exists.
-     * </li>
+     * <ul>
+     * <li>The first set contains fixed (statically) configured project to be trigger.</li>
+     * <li>The second set contains dynamically configured project, resolved by back tracking builds environment variables.</li>
+     * <li>The third set contains other recently triggered project found during back tracking builds</li>
+     * <li>The fourth set contains dynamically configured project that couldn't be resolved or project that doesn't exists.</li>
+     * </ul>
```
Comment (recena) [inline src/main/java/hudson/plugins/parameterizedtrigger/BuildTriggerConfig.java]: I love these changes. Details are very important. Thanks.

File: pom.xml [hunk 3]
```diff
@@ -36,34 +36,43 @@
         <dependency>
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>subversion</artifactId>
-            <version>1.38</version>
-            <optional>true</optional>
```
Comment (mslusarczyk) [inline pom.xml]: not optional anymore?

Comment (armfergom) [inline pom.xml]: The dependency is used in compilation time so I would assume it should not be optional. I've looked at the commit that made the dependency optional, and it was the only change between versions 2.5 and 2.6, so there may be other implications I have not thought of. For that reason I have made it optional again :) Thanks for bringing it up!

Comment (mslusarczyk) [inline pom.xml]: No problem:) You removed optional in one or two other places in this PR.

File: pom.xml [hunk 4]
```diff
@@ -36,34 +36,44 @@
...
             <artifactId>subversion</artifactId>
-            <version>1.38</version>
+            <version>2.5.7</version>
             <optional>true</optional>
         </dependency>
         <dependency>
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>matrix-project</artifactId>
-            <version>1.4</version>
-            <optional>true</optional>
```
Comment (andresrc) [inline pom.xml]: This seems correct: https://github.com/jenkinsci/parameterized-trigger-plugin/blob/master/src/main/java/hudson/plugins/parameterizedtrigger/FileBuildParameters.java#L233

**Comment by andresrc** [comment]
:bee:

**Comment by armfergom** [comment]
@reviewbybees done

**Comment by armfergom** [comment]
Close and opening to trigger CI job again.

**Comment by jglick** [comment]
@svanoort you still maintaining this?

**Comment by jglick** [comment]
I am not at all interested. :-)

**Comment by kwhetstone** [comment]
Looks like all the checks have passed.  Is it possible to get this merged an a new version of the plugin cut today?  Anything to get the various test harnesses up and running.

**Pull request merged**